TF/src/common/Storage.py

```diff
@@ -0,0 +1,208 @@
+                    src_node_index = queue.get()
+
+                    if src_node_index not in adj_list:
+                        continue
+
+                    for [edge_index, dest_node_index] in adj_list[src_node_index]:
+
+                        if edge_index not in from_nodes:
+                            from_nodes.update({edge_index : set()})
+                        
+                        from_nodes[edge_index].add(src_node_index + 1)
+
+                        if edge_index not in to_nodes:
+                            to_nodes.update({edge_index : set()})
+
+                        to_nodes[edge_index].add(dest_node_index + 1)
+
+                        if not vis[dest_node_index]:
+                            vis[dest_node_index] = True
+                            queue.put(dest_node_index)
+
+        # Surrogate Id for tensors
+        tensor_id = 0
+
+        # Number of edges to be processed per batch
+        num_attributes = len(vars(graph.edges[0]))
+        num_edges_per_batch = 20000 // num_attributes
+        
+        edge_indices = list(to_nodes.keys())
+        num_edges = len(edge_indices)
+
+        # Inserting data into the Tensors table
+        # Looping till all edges are inserted
+        while tensor_id < num_edges:
+            # Creating and inserting a batch
+            with self.database.batch() as batch:
+                for _ in range(num_edges_per_batch):
+                    if tensor_id == num_edges:
+                        return
+
+                    edge_index = edge_indices[tensor_id]
+                    edge = graph.edges[edge_index]
+                    to_operator_ids = list(to_nodes[edge_index])
+                    from_operator_ids = list(from_nodes[edge_index])
+
+                    # To store the database column names and their 
+                    # values to be inserted
+                    columns = [
+                        'model_name', 'tensor_id', 
+                        'from_operator_ids', 'to_operator_ids'
+                        ]
+                    values = [
+                        graph.model_name, tensor_id + 1, 
+                        from_operator_ids, to_operator_ids
+                        ]
+
+                    # Adding Edge attributes and their values to columns and values
+                    attrs = vars(edge)
+                    for item in attrs.items():
+                        if item[0] != 'label' and item[0] != 'value':
+                            columns.append(item[0])
+                            values.append(item[1])
+
+                    columns = tuple(columns)
+                    values = tuple(values)
+
+                    # Insert 
+                    batch.insert(
+                        table = 'Tensors',
+                        columns = columns,
+                        values = [values]
+                    )
+                    tensor_id += 1
+
+    # Function to load data for given Graph to Spanner Database
```

Comment:
Why not pull all common stuff in to a base class and the derive it for both TF and TFLite. Again lot of code duplication.

Reply:
Fixed in commit with message 'Restructure code', changed file structure for common files.